hw/ip/uart/data/uart.hjson

```diff
@@ -64,7 +64,7 @@
     { name: "rx_watermark",
       type: "status",
       desc: "raised if the receive FIFO is past the high-water mark."}
-    { name: "tx_empty",
+    { name: "tx_done",
       desc: "raised if the transmit FIFO has emptied and no transmit is ongoing."}
```

Comment:
I do not think we need further last-minute changes.  But I was trying to think how my code would use this, if trying to output "Rebooting..." on the console before doing a reset.

Naively, I would make sure to clear this interrupt status bit if set, then run a loop that puts the characters into the FIFO (while checking TXFULL bit), and then I would naively wait until seeing this tx_done status bit/interrupt before performing reset.

However, I came to think that if some other higher-priority interrupt happens during the loop, the FIFO could run empty and the tx_done flag be latched, before the loop continues, and then when the loop is done, my logic above would think that transmission is done when really the last half of the characters are still being actively output.  I would like to avoid having to clear the bit and put things into the FIFO while disabling interrupts, in order to guarantee that the FIFO cannot run empty "in the middle" of the operation.

As far as I can see, the above could be avoided if either this event type interrupt bit was automatically cleared upon writing any character to the TX FIFO, or if it too became a status type interrupt, which would be active exactly when the TX FIFO is empty and no transmit is ongoing.

As I said, I do not think the Ti50 team needs this change.  But I am just wondering how we are anticipating it could be used.